mathics/builtin/numeric.py

```diff
@@ -824,7 +825,437 @@ def apply(self, n, base, evaluation, nr_elements=None):
+    #> RealDigits[1/197, 260, 5]
+     = {{1, 83, 38, 71, 69}, 0}
+     
+    #> RealDigits[1/197, 260, 5, -6]
+     = {{246, 208, 137, 67, 80}, -5}
+     
+    #> RealDigits[Pi, 260, 20]
+     = {{3, 36, 211, 172, 124, 173, 210, 42, 162, 76, 23, 206, 122, 187, 23, 245, 241, 225, 254, 98}, 1}
+  
+    #> RealDigits[Pi, 260, 5]
+     = {{3, 36, 211, 172, 124}, 1}
+     
+    #> RealDigits[1/3]
+     = {{{3}}, 0}
+     
+    #> RealDigits[1/2, 7]
+     = {{{3}}, 0}
+     
+    #> RealDigits[3/2, 7]
+     = {{1, {3}}, 1}
+    
+    #> RealDigits[-3/2, 7]
+     = {{1, {3}}, 1}
+     
+    #> RealDigits[3/2, 6]
+     = {{1, 3}, 1}
+     
+    #> RealDigits[1, 7, 5]
+     = {{1, 0, 0, 0, 0}, 1}
+     
+    #> RealDigits[I, 7]
+     : The value I is not a real number.
+     = RealDigits[I, 7]
+     
+    #> RealDigits[-Pi]
+     : The number of digits to return cannot be determined.
+     = RealDigits[-Pi]
+     
+    #> RealDigits[Round[x + y]]
+     = RealDigits[Round[x + y]]
+    
+    """
+    attributes = ('Listable',)
+    
+    messages = {
+        'realx': 'The value `1` is not a real number.',
+        'ndig': 'The number of digits to return cannot be determined.',
+        'rbase': 'Base `1` is not a real number greater than 1.',
+        'intnm': 'Non-negative machine-sized integer expected at position 3 in `1`.',
+        'intm' : 'Machine-sized integer expected at position 4 in `1`.',
+    }
+    
+    def apply_complex(self, n, var, evaluation):
+        'RealDigits[n_Complex, var___]'
+        return evaluation.message('RealDigits', 'realx', n)
+    
+    def apply_rational_with_base(self, n, b, evaluation):
+        'RealDigits[n_Rational, b_Integer]'
+        
+        expr = Expression('RealDigits', n)
+           
+        py_n = abs(n.value)
+        py_b = b.to_python()
+        
+        if check_finite_decimal(n.denominator().to_python()) == True and not py_b % 2 :
+            return self.apply_2(n, b, evaluation)
+        else:
+            exp = int(mpmath.ceil(mpmath.log(py_n, py_b)))
+            (head, tails) = convert_repeating_decimal(py_n.as_numer_denom()[0], py_n.as_numer_denom()[1], py_b)
+            
+            list_str = Expression('List')
+            for x in head:
+                if not x == '0':
+                    list_str.leaves.append(from_python(int(x)))
+            list_str.leaves.append(from_python(tails))
+            
+        return Expression('List', list_str, exp)
+    
+    def apply_rational_without_base(self, n, evaluation):
+        'RealDigits[n_Rational]'
+
+        return self.apply_rational_with_base(n, from_python(10), evaluation)
+ 
+    def apply(self, n, evaluation):
+        'RealDigits[n_]'
+        
+        # Handling the testcases that throw the error message and return the ouput that doesn't include `base` argument
+        if isinstance(n, Symbol):
+            return evaluation.message('RealDigits', 'ndig', n) if n.name.startswith('System`')  else Expression('RealDigits', n)
```

Comment:
better to return `None` in the else case here to prevent re-evaluating the result.

```
if isinstance(n, Symbol) and n.get_name().startswith('System`'):
    return evaluation.message('RealDigits', 'ndig', n)
```